### DISCUSSION TITLE
Nextflow error - Capsule

### DISCUSSION THREAD

**Discussion by hmmachko** [nextflow-io/nextflow#4430] [Q&A]
### Expected behavior and actual behavior
I am attempting to use nextflow for viralrecon on my Mac OS local linux terminal. I am unable to initialize the nextflow environment du to Capsule parse version error.

### Steps to reproduce the problem
I downloaded nextflow with:
`wget -qO- https://get.nextflow.io | bash`
Any command to try and use nextflow, such as `nextflow -h` yields the text shown below in Program output. 

###Program output
`CAPSULE EXCEPTION: Could not parse version: 18.0.1.1 while processing attribute Min-Java-Version: 1.8.0 (for stack trace, run with -Dcapsule.log=verbose)
USAGE: java <options> -jar .nextflow/framework/22.04.5/nextflow-22.04.5-one.jar`

###Environment
Nextflow version: 23.10.1 (installed 2013-10-17)
Java version:
openjdk version "17.0.6" 2023-01-17
OpenJDK Runtime Environment Temurin-17.0.6+10 (build 17.0.6+10)
OpenJDK 64-Bit Server VM Temurin-17.0.6+10 (build 17.0.6+10, mixed mode, sharing)
Operating system: Mac OS 13.2.1 
Bash version: 5.8.1 (x86_64-apple-darwin22.0)

**Comment by hmmachko** [comment]
Sorry additional and corrected information:
`nextflow -h -Dcapsule.log=verbose
CAPSULE: Jar: /Users/labadmin/.nextflow/framework/22.04.5/nextflow-22.04.5-one.jar
CAPSULE: Platform: macos
CAPSULE: Loading caplet: MavenCapsule
CAPSULE: Applying caplet MavenCapsule
CAPSULE: Initializing app ID
CAPSULE: Initialized app ID: nextflow_22.04.5
CAPSULE: Launching app nextflow_22.04.5
CAPSULE: Cleanup
CAPSULE EXCEPTION: Could not parse version: 18.0.1.1 while processing attribute Min-Java-Version: 1.8.0
java.lang.IllegalArgumentException: Could not parse version: 18.0.1.1`

**Comment by pditommaso** [comment]
There's something messed in your configuration. You are reporting the use of Java 17.0.6 and nextflow 23.10.1. 

However the logs says Nextflow 22.04.5 and Java 18.0.1.1

**Comment by colindaven** [comment]
I had this error quite a lot about 18 months ago but solved it by using `java=11.0.1` in a conda environment.

On Linux I also had to unset any conflicting java env settings and potentially also reset JAVA_HOME to the one I wanted to use

```
# Setup java env 
unset JAVA_HOME
unset _JAVA_OPTIONS
unset JAVA_OPTIONS
```

You can check your java envs on Linux with 

`env | grep JAVA`
